### PR TITLE
Implement invoke

### LINKED ISSUE #208: Parser error messages
The parser gives bad error messages when:

- [x] When a cell is missing a semicolon
- [x] When numbers without bitwidths are used.
- [x] When `while` bodies don't have `seq` inside them.
- [x] When you have a non-binary number in a binary literal like in `32'b4`

### PR DESCRIPTION
**Pull request by rachitnigam**
Basic implementation for `invoke`. 

TODO:
- [x] Implement compilation for invoke
- [ ] Remove the `unimplemented!()` macro calls.
  - Punt this to the next PR.

### CODE REVIEW

**Comment by rachitnigam** [comment]
Ready to demo this today!

**Comment by rachitnigam** [comment]
I feel that generating the `once` register is unnecessary. The control logic that calls the generated `invoke` group immediately shuts off the execution once the done signal is high. Furthermore, the control logic also makes sure that the path from the `go` signal to the `done` signal has a register in between.

TLDR: Generation can probably just be
```
group invoke {
  comp.go = 1'd1;
  comp.arg1 = ...;
  invoke[done] = comp.done;
}
```

**Comment by rachitnigam** [comment]
This is ready to merge. Waiting on #295.

**Comment by sampsyo** [comment]
Woohoo! Long live `invoke`!
